Mage.Sets/src/mage/cards/c/ChissGoriaForgeTyrant.java

```diff
@@ -0,0 +1,204 @@
+        super(card);
+    }
+
+    @Override
+    public ChissGoriaForgeTyrant copy() {
+        return new ChissGoriaForgeTyrant(this);
+    }
+}
+
+class ChissGoriaForgeTyrantEffect extends OneShotEffect {
+
+    ChissGoriaForgeTyrantEffect() {
+        super(Outcome.Benefit);
+        staticText = "exile the top five cards of your library. You may cast an artifact spell from among them this turn. If you do, it gains affinity for artifacts.";
+    }
+
+    private ChissGoriaForgeTyrantEffect(final ChissGoriaForgeTyrantEffect effect) {
+        super(effect);
+    }
+
+    @Override
+    public ChissGoriaForgeTyrantEffect copy() {
+        return new ChissGoriaForgeTyrantEffect(this);
+    }
+
+    @Override
+    public boolean apply(Game game, Ability source) {
+        Player player = game.getPlayer(source.getControllerId());
+        if (player == null) {
+            return false;
+        }
+
+        // Return if no cards exiled, effect still applied
+        Cards cards = new CardsImpl(player.getLibrary().getTopCards(game, 5));
+        if (cards.isEmpty()) {
+            return true;
+        }
+        player.moveCards(cards, Zone.EXILED, source, game);
+
+        Cards castableCards = new CardsImpl(cards.getCards(StaticFilters.FILTER_CARD_NON_LAND, game));
+        game.addEffect(new ChissGoriaForgeTyrantCanPlayEffect(castableCards, game), source);
+        return true;
+    }
+}
+
+class ChissGoriaForgeTyrantCanPlayEffect extends AsThoughEffectImpl {
+
+    private final Set<MageObjectReference> morSet = new HashSet<>();
+
+    ChissGoriaForgeTyrantCanPlayEffect(Cards cards, Game game) {
+        super(AsThoughEffectType.PLAY_FROM_NOT_OWN_HAND_ZONE, Duration.EndOfTurn, Outcome.Benefit);
+        cards.stream().map(uuid -> new MageObjectReference(uuid, game)).forEach(morSet::add);
+    }
+
+    private ChissGoriaForgeTyrantCanPlayEffect(final ChissGoriaForgeTyrantCanPlayEffect effect) {
+        super(effect);
+        this.morSet.addAll(effect.morSet);
+    }
+
+    @Override
+    public ChissGoriaForgeTyrantCanPlayEffect copy() {
+        return new ChissGoriaForgeTyrantCanPlayEffect(this);
+    }
+
+    @Override
+    public boolean apply(Game game, Ability source) {
+        return true;
+    }
+
+    @Override
+    public boolean applies(UUID sourceId, Ability source, UUID affectedControllerId, Game game) {
+        if (!source.isControlledBy(affectedControllerId)) {
+            return false;
+        }
+        UUID objectIdToCast = CardUtil.getMainCardId(game, sourceId);
+        return game.getCard(objectIdToCast).isArtifact(game) 
+            && morSet.stream().anyMatch(mor -> mor.refersTo(objectIdToCast, game))
+            && ChissGoriaForgeTyrantWatcher.checkRef(source, morSet, game);
+    }
+}
+
+// TODO: Unsure where to apply this effect (or AffinityForArtifactsAbility) to make it work.
+class ChissGoriaForgeTyrantCostEffect extends CostModificationEffectImpl {
```

Comment:
Cost modification isn't technically correct, we want to add the existing Affinity for artifacts ability